tests/unit/test_auth.py

```diff
@@ -1,7 +1,24 @@
 import unittest
+import pytest
+from typing import Optional
+from unittest.mock import patch
 
-from databricks.sql.auth.auth import AccessTokenAuthProvider, BasicAuthProvider, AuthProvider
+from databricks.sql.auth.auth import AccessTokenAuthProvider, BasicAuthProvider, AuthProvider, DatabricksOAuthProvider
 from databricks.sql.auth.auth import get_python_sql_connector_auth_provider
+from databricks.sql.auth.oauth import OAuthManager
+from databricks.sql.auth.endpoint import OAuthEndpointsAzure, OAuthEndpointsAws, CloudType
+from databricks.sql.experimental.oauth_persistence import OAuthPersistence, OAuthToken
+
+
+class OAuthPersistenceCache(OAuthPersistence):
```

Comment:
Is there any reason we _shouldn't_ just make this type of cache available to end-users @moderakh? Right now our examples simply tell users to roll their own cache. This in-memory demo could be useful for anyone starting out.

If not, I think we may as well name it `TestOAuthPersistenceCache` to make it obvious this is only for testing purposes.

Reply:
This is only for unit test. Yes, we can name it to `TestOAuthPersistenceCache`

Reply:
I moved `OAuthPersistenceCache` to `oauth_persistence.py` so it can also be reused by others.

Reply:
SGTM